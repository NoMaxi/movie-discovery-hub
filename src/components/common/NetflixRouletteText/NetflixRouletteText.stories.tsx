import type { Meta, StoryObj } from "@storybook/react";
import { NetflixRouletteText } from "./NetflixRouletteText";

const meta = {
    title: "Common/NetflixRouletteText",
    component: NetflixRouletteText,
} satisfies Meta<typeof NetflixRouletteText>;

export default meta;

type Story = StoryObj<typeof NetflixRouletteText>;

export const Default: Story = {};
