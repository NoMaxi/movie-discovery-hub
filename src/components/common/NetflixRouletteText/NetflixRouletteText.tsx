export const NetflixRouletteText = () => (
    <span className="text-[var(--color-primary)] group">
        <span className="font-black group-hover:text-blue-500 transition-colors duration-300">netflix</span>
        <span className="font-medium group-hover:text-red-500 transition-colors duration-300">roulette</span>
    </span>
);
