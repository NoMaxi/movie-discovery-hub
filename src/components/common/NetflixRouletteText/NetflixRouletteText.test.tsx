import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { NetflixRouletteText } from "./NetflixRouletteText";

describe("NetflixRouletteText", () => {
    test("Should render the component with correct text parts", () => {
        const { asFragment } = render(<NetflixRouletteText />);
        const netflixPart = screen.getByText("netflix");
        const roulettePart = screen.getByText("roulette");

        expect(netflixPart).toBeInTheDocument();
        expect(roulettePart).toBeInTheDocument();
        expect(asFragment()).toMatchSnapshot();
    });

    test("Should apply correct CSS classes to elements", () => {
        render(<NetflixRouletteText />);
        const parentSpan = screen.getByTestId("netflix-roulette-text");
        const netflixSpan = screen.getByText("netflix");
        const rouletteSpan = screen.getByText("roulette");

        expect(parentSpan).toContainElement(netflixSpan);
        expect(parentSpan).toContainElement(rouletteSpan);
        expect(parentSpan).toHaveClass("text-primary group");
        expect(netflixSpan).toHaveClass("font-black group-hover:text-blue-500 transition-colors duration-300");
        expect(rouletteSpan).toHaveClass("font-medium group-hover:text-red-500 transition-colors duration-300");
    });
});
