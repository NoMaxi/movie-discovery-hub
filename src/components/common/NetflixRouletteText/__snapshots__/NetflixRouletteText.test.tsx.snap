// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NetflixRouletteText Should render the component with correct text parts 1`] = `
<DocumentFragment>
  <span
    class="text-[var(--color-primary)] group"
    data-testid="netflix-roulette-text"
  >
    <span
      class="font-black group-hover:text-blue-500 transition-colors duration-300"
    >
      netflix
    </span>
    <span
      class="font-medium group-hover:text-red-500 transition-colors duration-300"
    >
      roulette
    </span>
  </span>
</DocumentFragment>
`;
